src/httpSpansAgent.js

```diff
@@ -69,18 +72,34 @@ export const HttpSpansAgent = (() => {
     sessionInstance = undefined;
   };
 
+  const sendHttpRequest = async (url, headers, requestBody) => {
+    const session = getSessionInstance();
+    let requestTimeoutTimer;
+    const requestTimeout = new Promise(resolve => {
+      requestTimeoutTimer = setTimeout(() => {
+        clearTimeout(requestTimeoutTimer);
+        logger.debug(
+          `Edge connection timeout [${CONNECTION_TIMEOUT}ms] from setTimeout (Tracer skipping)`
+        );
+        resolve(1);
+      }, CONNECTION_TIMEOUT);
+    });
+    const requestPromise = session.post(url, requestBody, { headers }).catch(e => {
+      logger.debug('Edge error (Tracer skipping)', e.message);
+    });
+
+    await Promise.race([requestPromise, requestTimeout]).finally(() => {
```

Comment:
isn't that will leave something in the event loop - which will cause the lambda to not finish? (in the case that the relevant flag is on)

Reply:
why don't we use a package that gives better timeout handling ?

Reply:
Axios timeout handling is not good enough (and is considered as the best), And about the event loop, its a real edge case that our edge will take more than 300ms, it's okay to leave it

Reply:
I disagree here-
if we will do a bad deployment that will crush the edge (or make it to be super slow)- then you may fill the customer's event loop with our events. So if his lambda has many warm requests, then we will kill his production :-(

Reply:
I disagree - we always should consider the case that the edge misbehaves (this is the original cause of this ticket!).
If the lambda will have many warm requests then you'll fill the event loop.. :(